LogMonitor/src/LogMonitor/ConfigFileParser.cpp

```diff
@@ -640,6 +652,38 @@ AddNewSource(
     return true;
 }
 
+///
+/// Validates that when root directory is passed, includeSubdirectories is false
+///
+/// \param Attributes   An AttributesMap that contains the attributes of the new source objet.
+/// \return false when root directory is passed, includeSubdirectories = true. Otherwise, true </returns>
+bool ValidateDirectoryAttributes(_In_ AttributesMap &Attributes)
+{
+    const std::wstring key = L"directory";
+
+    if (!Utility::ConfigAttributeExists(Attributes, JSON_TAG_DIRECTORY) ||
+        !Utility::ConfigAttributeExists(Attributes, JSON_TAG_INCLUDE_SUBDIRECTORIES))
+    {
+            return true;
+    }
+
+    std::wstring directory = *(std::wstring *)Attributes[JSON_TAG_DIRECTORY];
+    const bool includeSubdirectories = *(bool *)Attributes[JSON_TAG_INCLUDE_SUBDIRECTORIES];

```

Comment:
Is includeSubDirectories a required field and what does it default to  if nothing is passed?

Reply:
`includeSubDirectories` is an optional field. It defaults to `false` when not provided. [LoggerSettings.h#l25](https://github.com/microsoft/windows-container-tools/blob/a47d40e76f63d8ad0f890b261783b862616fcd6a/LogMonitor/src/LogMonitor/Parser/LoggerSettings.h#L257C5-L257C40)